sw/device/tests/dif/dif_clkmgr_smoketest.c

```diff
@@ -0,0 +1,95 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/base/memory.h"
+#include "sw/device/lib/dif/dif_clkmgr.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/test_main.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"  // Generated.
+
+const test_config_t kTestConfig = {0};
+
+/**
+ * Test that all 'gateable' clocks, directly controlled by software,
+ * can be enabled and disabled.
+ */
+static void test_gateable_clocks(const dif_clkmgr_t *clkmgr) {
+  const dif_clkmgr_gateable_clock_t clocks[] = {
+      kTopEarlgreyGateableClocksIoDiv4Peri,
+      kTopEarlgreyGateableClocksUsbPeri,
+  };
+
+  for (int i = 0; i < ARRAYSIZE(clocks); ++i) {
+    dif_clkmgr_gateable_clock_t clock = clocks[i];
+    bool enabled;
```

Comment:
nit: You may want to add a comment saying this is not checked after the call because you are testing the behavior not the current (reset) value.

Reply:
Done